HIP/hip-583.md

```diff
@@ -10,220 +10,451 @@ status: Accepted
+#### Validation of the `alias`
 
-`CryptoTransfer` transactions should also support this case.
-This will support users, developers and exchanges who prefer to create and credit an accounts balance in one step.
+It **is** possible to specify an `account-num alias` as the _target_ of a `CryptoTransfer` transaction. Indeed, for
+Ethereum-native wallets, this is the only way to transfer value to Hedera accounts that don't have an
+`evm address alias`. However, it is **impossible** to auto-create an account using the `account-num alias`. If the
+account does not exist, the transfer will fail.
 
-`EthereumTransaction` transactions where value is transfered  (i.e `tx.to` and `tx.value` are set) should also support this case.
-This will support users of `Ethereum-native wallets` and eth dev tools who are accumstomed to creating and crediting an accounts balance in one step.
-In this case th `tx.to` value of the RLP encoded transaction would dictate the `public-address` to be used as an alias.
+##### `key alias`
 
-### Allow CryptoCreate to automatically set alias
+The `key alias` is a protobuf-encoded `Key` object. We only support `ED25519` and `ECDSA_secp256k1` keys at this time.
+When an attempt is made to create a new account with `CryptoCreate` with an alias, we check whether it is a `key alias`
+by checking if there are more than 20 bytes. We then:
 
-When `CryptoCreate.key` is an `ECDSA` key, the network node may calculate the `public-address` and automatically set the accounts alias value. 
+1. Deserialize the protobuf object. If the object is not a protobuf object, we reject the transaction.
+2. If the protobuf object is not an `ED25519` or `ECDSA_secp256k1` `Key`, we reject the transaction.
+3. If the `Key` is an `ED25519` key:
+    1. If the alias is already in use, we use that account and do not create a new one
+    2. If the alias is not in use, we create a new account and use the given key as the key on the account
+4. If the `Key` is an `ECDSA_secp256k1` key:
+    1. If the alias is already in use, we use that account and do not create a new one
+    2. If the derived `evm address` from the key is already in use, we use that account and do not create a new one
+    3. If the alias is not in use, we create a new account and use the given key as the key on the account.
 
-## Backwards Compatibility
+If we do create a new account from either `CryptoCreate` or `CryptoTransfer`, and the `Key` is an `ECDSA_secp256k1` key,
+then we reserve in the aliases for **both** the raw protobuf-encoded `Key` bytes **and** the derived evm address.
```

Comment:
If we create a new account with an `ECDSA_secp256k1` key, we no longer automatically reserve the raw protobuf-encoded `Key` bytes **and** the derived evm address aliases. Or do you mean only if the account has that key as an alias? In that case, it's correct.

Reply:
Yes, only in the case that the account has the alias.

Reply:
Updating the language:

>If we do create a new account from either `CryptoCreate` or `CryptoTransfer`, and the `alias` is an `ECDSA_secp256k1` key, then we reserve in the aliases for **both** the raw protobuf-encoded `Key` bytes **and** the derived evm address. Essentially, this allows the account created with an `ECDSA_secp256k1` to be addressed by that alias, the `evm address alias`, and the `account-num alias`.

Reply:
Done